addons/mail/static/src/model/model_manager.js

```diff
@@ -1069,9 +1066,10 @@ export class ModelManager {
             relFunc(Model.name, { inverse: field.fieldName }),
             {
                 fieldName: `_inverse_${Model}/${field.fieldName}`,
-                // Note that an identifying field is not necessarily defined as
-                // `required` (example: identifyingFields with an OR). 
-                isCausal: field.required || Model.__identifyingFieldsFlattened.has(field.fieldName),
+                // Add isCausal to inverses that are identifying fields. This is
+                // needed to support models that aren't directly defined in
+                // JavaScript.
```

Comment:
I feel like a comment is still relevant here. If I wasn't the one who wrote that code, I probably wouldn't have understood why this line is necessary. But apart from that I agree with you, it's too specific and it would be better to at least reword it to be more general.